fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/mv/SelectMaterializedIndexWithAggregate.java

```diff
@@ -106,545 +106,671 @@ public List<Rule> buildRules() {
                 // only agg above scan
                 // Aggregate(Scan)
                 logicalAggregate(logicalOlapScan().when(this::shouldSelectIndex)).thenApply(ctx -> {
-                    LogicalAggregate<LogicalOlapScan> agg = ctx.root;
-                    LogicalOlapScan scan = agg.child();
-                    SelectResult result = select(
-                            scan,
-                            agg.getInputSlots(),
-                            ImmutableSet.of(),
-                            extractAggFunctionAndReplaceSlot(agg, Optional.empty()),
-                            agg.getGroupByExpressions(),
-                            new HashSet<>(agg.getExpressions()));
-
-                    LogicalOlapScan mvPlan = scan.withMaterializedIndexSelected(result.preAggStatus, result.indexId);
-                    SlotContext slotContext = generateBaseScanExprToMvExpr(mvPlan);
-
-                    if (result.exprRewriteMap.isEmpty()) {
-                        return new ReplaceExpressions(slotContext)
-                            .replace(agg.withChildren(mvPlan), mvPlan);
-                    } else {
-                        return new LogicalProject<>(
-                            generateProjectsAlias(agg.getOutputs(), slotContext),
-                                new ReplaceExpressions(slotContext).replace(
-                                    new LogicalAggregate<>(
-                                        agg.getGroupByExpressions(),
-                                        replaceAggOutput(
-                                            agg, Optional.empty(), Optional.empty(), result.exprRewriteMap),
-                                        agg.isNormalized(),
-                                        agg.getSourceRepeat(),
-                                        mvPlan
-                                    ), mvPlan));
+                    try {
```

Comment:
i think the better way is add a new interface `thenApplyNoThrow`. do try catch in it